credential-generator/program-cert-pre-processor/src/main/scala/org/sunbird/job/programcert/functions/ProgramCertPreProcessorFn.scala

```diff
@@ -154,6 +160,12 @@ class ProgramCertPreProcessorFn(config: ProgramCertPreProcessorConfig, httpUtil:
               logger.info("Adding the kafka event for programId: " + courseParentId)
               createIssueCertEventForProgram(courseParentId, event.userId, batchId, context)(metrics)
             }
+            if (!isSkippedEvent) {
+              //Add kafka event to failed to generate Certificate for Program
+              logger.info("Adding failed kafka event for programId: " + courseParentId)
+              context.output(config.generateCertificateFailedOutputTag, event)
+              metrics.incCounter(config.programCertIssueEventsCount)
+            }
```

Comment:
not correct, the intention is to not to generate program certification event instead add this event to failed topic - so that we can push these events at mid night on the same day and see... 

let me know this makes sense... 

if (isProgramCertificateToBeGenerated) {
	if (isFailedEvent) {
		//Add kafka event to failed to generate Certificate for Program
		logger.info("Program cert validation failed for event : " + event)
		context.output(config.generateCertificateFailedOutputTag, event)
		metrics.incCounter(config.programCertIssueEventsCount)
	} else {
		//Add kafka event to generate Certificate for Program
		logger.info("Adding the kafka event for programId: " + courseParentId)
		createIssueCertEventForProgram(courseParentId, event.userId, batchId, context)(metrics)
	}
}